shim/shim.c

```diff
@@ -296,25 +297,67 @@ handle_data(int infd, int outfd)
 	}
 }
 
+/*!
+ * Print help usage
+ *
+ * \param prog Binary name
+ */
+void print_usage(char *prog) {
+	printf("%s: Usage\n", prog);
+        printf("  -c,  --container_id   Container id\n");
+        printf("  -p,  --proxy_sock_fd  File descriptor of the socket connected to cc-proxy\n");
+        printf("  -h,  --help           Display this help message\n");
+}
+
 int
 main(int argc, char **argv)
 {
 	int proxy_sockfd;
 	struct sockaddr_un proxy_address;
-	char* container_id;
+	char* container_id = NULL;
 	char* proxy_sock_path;
 	struct pollfd poll_fds[MAX_POLL_FDS];
 	nfds_t nfds = 0;
 	int ret;
 	struct sigaction sa;
+	int c;
+	int proxy_sock_fd = -1;
+	char *endptr;
+
+	struct option prog_opts[] = {
+		{"container_id", required_argument, 0, 'c'},
+		{"proxy_sock_fd", required_argument, 0, 'p'},
+		{"help", no_argument, 0, 'h'},
+		{ 0, 0, 0, 0},
+	};
+
+	while ((c = getopt_long(argc, argv, "c:p:h", prog_opts, NULL))!= -1) {
+		switch (c) {
+			case 'c':
+				container_id = strdup(optarg);
+				break;
+			case 'p':
+				if ((ret = (int)strtol(optarg, &endptr, 10)) == -1 || *endptr != '\0') {
```

Comment:
I'd break this line up into separate statements as that would make it clearer.

Also, `strtol()` doesn't return `-1` on error - you need to check for `LONG_MIN` and `LONG_MAX`. However, it would be just as good to set `errno=0` before the call and just check `(!*endptr && !errno)` after it rather than checking `ret` directly.